src/components/atoms/AvatarMenu/AvatarMenu.tsx

```diff
@@ -0,0 +1,63 @@
+import Avatar from '@material-ui/core/Avatar'
+import IconButton from '@material-ui/core/IconButton'
+import Menu from '@material-ui/core/Menu'
+import MenuItem from '@material-ui/core/MenuItem'
+import * as React from 'react'
+
+import { Images } from 'images/Images'
+
+import { User } from 'interfaces/Props'
+
+interface AvatarMenuProps {
+  readonly user?: User
+  readonly onSignOut?: () => any
+}
+
+interface AvatarMenuState {
+  anchorEl: object
+}
+
+export class AvatarMenu extends React.Component<AvatarMenuProps, AvatarMenuState> {
+  state: { anchorEl: any } = {
+    anchorEl: null,
+  }
+
+  handleClick = (event: any) => {
+    this.setState({ anchorEl: event.currentTarget })
```

Comment:
We should use the functional form of setState so that our state updates are always deterministic.
See https://medium.com/javascript-scene/setstate-gate-abc10a9b2d82

Reply:
Is this still true even though we aren't concerned about the previous state?

Reply:
I'm thinking more in terms of the whole codebase rather than this particular usage.
We want to adopt a consistent way of using setState throughout the codebase. By using the functional form, we avoid any instance, now or in the future, where non-deterministic behavior is the root of a bug. The time and expense are completely avoided by choosing the functional form.

Reply:
I think there is another possibility to handle states with the render props.
I talked about it with @lautarodragan 2 months ago

I leave  this article 

https://reactjs.org/docs/render-props.html

Reply:
I updated it to just the functional setState for this PR. I've created another branch and added an AnchorProvider Render Prop Component. Once we merge these 2 PR's in I will create that PR.